azurerm/internal/services/storage/resource_arm_hpc_cache_nfs_target.go

```diff
@@ -0,0 +1,288 @@
+package storage
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+
+	"github.com/Azure/azure-sdk-for-go/services/storagecache/mgmt/2019-11-01/storagecache"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/parsers"
+	storageValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func namespaceJunctionResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"namespace_path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: storageValidate.HPCCacheNamespacePath,
+			},
+			"nfs_export": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: storageValidate.HPCCacheNFSExport,
+			},
+			"target_path": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "",
+				ValidateFunc: storageValidate.HPCCacheNFSTargetPath,
+			},
+		},
+	}
+}
+func resourceArmHPCCacheNFSTarget() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHPCCacheNFSTargetCreateOrUpdate,
+		Update: resourceArmHPCCacheNFSTargetCreateOrUpdate,
+		Read:   resourceArmHPCCacheNFSTargetRead,
+		Delete: resourceArmHPCCacheNFSTargetDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parsers.HPCCacheTargetID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: storageValidate.HPCCacheTargetName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"cache_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"host_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"usage_model": {
```

Comment:
could we order these alphabetically?

Reply:
I thought of to order them in alphabet, while I thought `usage_model` is a less frequently changed field compared to `namespace_junction`, that's why I put the latter later. However, I reconsider now and find it might not be true. So we can just order it per your suggestion.